src/OpenTelemetry/Logs/LogRecord.cs

```diff
@@ -81,21 +107,97 @@ public sealed class LogRecord
 
         public EventId EventId { get; }
 
-        public string FormattedMessage { get; }
+        /// <summary>
+        /// Gets or sets the FormattedMessage of the log.
+        /// </summary>
+        public string FormattedMessage
+        {
+            get => this.formattedMessage;
+            set
+            {
+                Guard.ThrowIfNullOrEmpty(value, nameof(this.FormattedMessage));
+                this.formattedMessage = value;
+            }
+        }
 
         /// <summary>
-        /// Gets the raw state attached to the log. Set to <see
+        /// Gets or sets the raw state attached to the log. Set to <see
         /// langword="null"/> when <see
         /// cref="OpenTelemetryLoggerOptions.ParseStateValues"/> is enabled.
         /// </summary>
-        public object State { get; }
+        public object State
+        {
+            get => (object)this.state;
```

Comment:
When upcasting back to object, it is no longer of type `FormattedLogValues` can cannot leverage its `.ToString` anymore...